### PR TITLE
Use custom docker image

### PR DESCRIPTION
**Pull request by marians**
- Use a custom docker image which is 6.8 MB in size
- Uses bash instead of sh for more predictable execution

### CODE REVIEW

**Review by pipo02mix — COMMENTED** [review]
nice

File: helm/app-updater-app/templates/cronjob.yaml [hunk 1]
```diff
@@ -15,9 +15,9 @@ spec:
           serviceAccount: app-updater
           restartPolicy: Never
           containers:
-          - image: teemow/debug
+          - image: quay.io/giantswarm/debug:update
```
Comment (pipo02mix) [inline helm/app-updater-app/templates/cronjob.yaml:18]: hmm what about use semver in the tagging?

Comment (marians) [reply helm/app-updater-app/templates/cronjob.yaml:18]: Will change to a specific image, as Timo suggested

**Comment by marians** [comment]
Testing the complete app now in gollum

**Comment by marians** [comment]
Currently fails with `ContainerCannotRun`: `OCI runtime create failed: container_linux.go:348: starting container process caused "exec: \"/opt/script/script.sh\": permission denied": unknown`

**Comment by pipo02mix** [comment]
What about add permissions in dockerfile? `chmod +x`

**Comment by marians** [comment]
I finally got it working with this combination:

Dockerfile:

```dockerfile
ENTRYPOINT ["/bin/bash"]
```

Cronjob:

```yaml
- image: "quay.io/giantswarm/app-updater:{{ .Values.image.version }}"
  name: app-updater
  args: ["/opt/script/script.sh"]
```

File: helm/app-updater-app/templates/configmap.yaml [hunk 2]
```diff
@@ -6,11 +6,13 @@ metadata:
...
 data:
   script.sh: |
+    #!/bin/bash
+    
     export KUBE_TOKEN=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
 
     export DSOURCE=$(curl -sSk -H "Authorization: Bearer $KUBE_TOKEN"  https://$KUBERNETES_SERVICE_HOST:$KUBERNETES_PORT_443_TCP_PORT/apis/application.giantswarm.io/v1alpha1/namespaces/{{ .Values.clusterID }}/apps/$APP_NAME | jq .spec.version -r)
 
-    export DTARGET=$(curl --silent "https://api.github.com/repos/$APP_REPO/releases/latest" | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/' | sed 's/^v//')
+    export DTARGET=$(curl --silent "https://api.github.com/repos/$APP_REPO/releases/latest"  | jq -r .tag_name | sed 's/^v//')
```
Comment (marians) [inline helm/app-updater-app/templates/configmap.yaml:15]: Just simplifying the script a bit here

File: helm/app-updater-app/templates/configmap.yaml [hunk 3]
```diff
@@ -19,8 +21,12 @@ data:
...
     rx='^([0-9]+\.){0,2}(\*|[0-9]+)$'
     if [[ $DSOURCE != $DTARGET ]] && [[ $DTARGET =~ $rx ]]; then
-        echo "App does need to be upgraded, proceeding to update App CR version field"
-        curl -sSk -o /dev/null -w "%{http_code} " -H "Authorization: Bearer $KUBE_TOKEN" --data "[{\"op\":\"replace\",\"path\":\"/spec/version\",\"value\":\"$DTARGET\"}]" -H "Content-Type: application/json-patch+json" -X PATCH https://$KUBERNETES_SERVICE_HOST:$KUBERNETES_PORT_443_TCP_PORT/apis/application.giantswarm.io/v1alpha1/namespaces/{{ .Values.clusterID }}/apps/$APP_NAME 
+        echo "App needs to be upgraded, proceeding to update App CR version field"
+        curl -sSk -o /dev/null -w "%{http_code} " \
+          -H "Authorization: Bearer $KUBE_TOKEN" \
+          -H "Content-Type: application/json-patch+json" \
+          -X PATCH https://$KUBERNETES_SERVICE_HOST:$KUBERNETES_PORT_443_TCP_PORT/apis/application.giantswarm.io/v1alpha1/namespaces/{{ .Values.clusterID }}/apps/$APP_NAME \
+          --data "[{\"op\":\"replace\",\"path\":\"/spec/version\",\"value\":\"$DTARGET\"}]"
```
Comment (marians) [inline helm/app-updater-app/templates/configmap.yaml:29]: Only adding some line breaks here for readability

File: helm/app-updater-app/templates/cronjob.yaml [hunk 4]
```diff
@@ -15,9 +15,9 @@ spec:
           serviceAccount: app-updater
           restartPolicy: Never
           containers:
-          - image: teemow/debug
+          - image: "quay.io/giantswarm/app-updater:{{ .Values.image.version }}"
             name: app-updater
-            command: ["/bin/sh", "/opt/script/script.sh"]
+            args: ["/opt/script/script.sh"]
```
Comment (pipo02mix) [inline helm/app-updater-app/templates/cronjob.yaml:20]: I guess this takes as command the entrypoint

**Review by pipo02mix — APPROVED** [review]
LGTM

**Pull request merged**